backend/common.js

```diff
@@ -52,15 +52,23 @@ export const handleRequest = async (req, res) => {
   workaroundForNodeMetrics(req);
   const k8sRequest = https
     .request(options, function(k8sResponse) {
+      if (
+        k8sResponse.headers['Content-Type'].includes('\\') ||
+        k8sResponse.headers['content-encoding'].includes('\\')
+      )
+        throw new Error('Response headers are potentially dangerous');
+
       res.writeHead(k8sResponse.statusCode, {
         'Content-Type': k8sResponse.headers['Content-Type'] || 'text/json',
         'Content-Encoding': k8sResponse.headers['content-encoding'] || '',
       });
-
       k8sResponse.pipe(res);
     })
     .on('error', function(err) {
       console.error('Internal server error thrown', err);
+
+      if (!err.match(/^[0-9a-zA-Z]$/)) err = '';
```

Comment:
You can try to return the error after it is sanitized.
You may check:
https://www.npmjs.com/package/string-sanitizer
or
https://developer.mozilla.org/en-US/docs/Web/API/Sanitizer/sanitizeToString

Reply:
I think we can allow the following chars `:{},"[]`. This will allow JSON error to be thrown which we handle on the frontend. What else do we possibly expect other than plain text or JSON? I think every other format of the error could be potentially malicious and there's no point to sanitize it.